File: rust/src/types/geometries/point.rs
use serde::{Serialize, Serializer};
use serde::ser::SerializeSeq;
use crate::types::geometries::coordinate::Coordinate;

#[derive(PartialEq, Clone, Debug)]
pub struct Point {
    pub coordinate: Coordinate,
}

impl Serialize for Point {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        let mut seq = serializer.serialize_seq(None)?;
        seq.serialize_element(&self.coordinate.x)?;
        seq.serialize_element(&self.coordinate.y)?;
        seq.end()
    }
}

File: java/mlt-core/src/main/java/org/springmeyer/Point.java
package org.springmeyer;

public class Point {
  public int x;
  public int y;

  public Point(int x, int y) {
    this.x = x;
    this.y = y;
  }

  public Point clone() {
    return new Point(this.x, this.y);
  }

  public String toString() {
    return "{x:" + this.x + ", y:" + this.y + "}";
  }
}
